automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala

```diff
@@ -244,6 +244,112 @@ class WorkspaceApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike
         }
       }
     }
+
+    "should not allow readers" - {
+      "to import method configs from another workspace" in {
+        withCleanBillingProject(owner) { projectName =>
+          withWorkspace(projectName, prependUUID("reader-import-config-dest-workspace"), aclEntries = List(AclEntry(studentA.email, WorkspaceAccessLevel.Reader))) { destWorkspaceName =>
+            withWorkspace(projectName, prependUUID("method-config-source-workspace"), aclEntries = List(AclEntry(studentA.email, WorkspaceAccessLevel.Reader))) { sourceWorkspaceName =>
+              withMethod("reader-import-from-workspace", MethodData.SimpleMethod) { methodName =>
+                val method = MethodData.SimpleMethod.copy(methodName = methodName)
+
+                val sourceMethodConfig = Map(
+                  "name" -> method.methodName,
+                  "namespace" -> method.methodNamespace,
+                  "workspaceName" -> Map(
+                    "namespace" -> projectName,
+                    "name" -> sourceWorkspaceName)
+                )
+
+                val destMethodConfig = Map(
+                  "name" -> s"destination-${method.methodName}",
+                  "namespace" -> s"destination-${method.methodNamespace}",
+                  "workspaceName" -> Map(
+                    "namespace" -> projectName,
+                    "name" -> destWorkspaceName)
+                )
+                Rawls.methodConfigs.createMethodConfigInWorkspace(
+                  projectName, sourceWorkspaceName, method, method.methodNamespace, method.methodName, 1,
+                  Map.empty, Map.empty, method.rootEntityType)(ownerAuthToken)
+
+                eventually {
+                  val copyFromWorkspaceException = intercept[RestException] {
+                    Rawls.methodConfigs.copyMethodConfigFromWorkspace(sourceMethodConfig, destMethodConfig)(studentAToken)
+                  }
+                  assertExceptionStatusCode(copyFromWorkspaceException, 403)
+                }
+              }(ownerAuthToken)
+            }(ownerAuthToken)
+          }(ownerAuthToken)
+        }
+      }
+
+      "to import method configs from the method repo" in {
+        withCleanBillingProject(owner) { projectName =>
+          withWorkspace(projectName, prependUUID("reader-import-config-dest-workspace"), aclEntries = List(AclEntry(studentA.email, WorkspaceAccessLevel.Reader))) { destWorkspaceName =>
+            withMethod("reader-import-from-method-repo", MethodData.SimpleMethod) { methodName =>
+              val method = MethodData.SimpleMethod.copy(methodName = methodName)
+
+              val destMethodConfig = Map(
+                "name" -> s"destination-${method.methodName}",
+                "namespace" -> s"destination-${method.methodNamespace}",
+                "workspaceName" -> Map(
+                  "namespace" -> projectName,
+                  "name" -> destWorkspaceName)
+              )
```

Comment:
`destMethodConfig` is defined in both tests, better to pull it out and put it either right above these two tests or the top of the class and only define it once

Reply:
`projectName` and `destWorkspaceName` are both only defined within fixtures in the test, so I think unfortunately I have to leave `destMethodConfig` defined in both tests. I could pull it out into a function if you think that'd be better